lib/Basic/Version.cpp

```diff
@@ -319,14 +319,16 @@ Optional<Version> Version::getEffectiveLanguageVersion() const {
   switch (Components[0]) {
   case 3:
 #ifdef SWIFT_VERSION_PATCHLEVEL
-    return Version{3, 2, SWIFT_VERSION_PATCHLEVEL};
+    return Version{3, 3, SWIFT_VERSION_PATCHLEVEL};
```

Comment:
@jrose-apple expressed some concern about moving this to 3.3 because comparisons on version ranges of being greater than 3.2 will encompass 4.0, 5.0, etc.  It's something we likely need to handle, but if people are using "> 3.2" to check if it is Swift 4 then bumping the Swift 3 version to "3.3.x" will cause problems.

Reply:
Thanks for bringing this up, Ted. I think I agree with Ewa that it's better to have something in place that we can modify later. We certainly don't want to continue advertising ourselves as Swift 3.2.

(Further information: we only allow >= checks, so the concern isn't about 3.2 but about 3.3. There's no way to check for "3.3 or 4.1 but not 4.0". This could also be a problem if we do any patch releases of 4.0.)